gpytorch/distributions/multitask_multivariate_normal.py

```diff
@@ -33,6 +34,19 @@ def __init__(self, mean, covariance_matrix, validate_args=False, interleaved=Tru
         if mean.dim() < 2:
             raise RuntimeError("mean should be a matrix or a batch matrix (batch mode)")
 
+        batch_shape = _mul_broadcast_shape(mean.shape[:-2], covariance_matrix.shape[:-2])
+        if mean.shape[-2:].numel() != covariance_matrix.size(-1):
+            if mean.size(-2) == 1:
+                mean = mean.expand(*batch_shape, covariance_matrix.size(-1) // mean.size(-1), mean.size(-1))
```

Comment:
Should we for good measure add a check that `covariance_matrix.size(-1)` is a multiple of `mean.size(-1)`? Otherwise this could result in some hard-to-debug errors...

Reply:
Yeah that makes sense. I'll update it

Reply:
Updated